### PR TITLE
Add out_dir to EvalHook

### PR DESCRIPTION
**Pull request by dumbPy**
Add `out_dir` to `EvalHook` so that the evaluation results can be dumped into a directory for further analysis like plotting confusion matrix.

Depending on automated comparison of a single metric is not always enough. This helps me plot my own graphs and confusion matrices.

### CODE REVIEW

File: mmaction/core/evaluation/eval_hooks.py [hunk 1]
```diff
@@ -60,6 +63,7 @@ def __init__(self,
                  by_epoch=True,
                  save_best='auto',
                  rule=None,
+                 out_dir=None,
```
Comment (innerlee) [inline mmaction/core/evaluation/eval_hooks.py:66]: How about saving it to work_dir, like the ckpts?

Comment (dreamerlin) [reply mmaction/core/evaluation/eval_hooks.py:66]: Also vote for saving to the `work_dir`

Comment (kennymckormick) [reply mmaction/core/evaluation/eval_hooks.py:66]: An advice: you can set the default value of `out_dir` as `work_dir`. For example, if  `out_dir is None`, you output the eval results to `work_dir`.

Comment (kennymckormick) [reply mmaction/core/evaluation/eval_hooks.py:66]: It's OK to save eval results by default, since it doesn't take much storage space.

Comment (innerlee) [reply mmaction/core/evaluation/eval_hooks.py:66]: Okay

**Comment by dreamerlin** [comment]
`EvalHook` has been moved to MMCV, so we will close this PR

**Pull request closed without merging**